meetings/2022-11-07.md

```diff
@@ -0,0 +1,107 @@
+* WT: I also went through most of the spec. I want to know what people see as possible obstacles.
+* ES: With migrating to OIDC Federation, what would the migration plan look like. The server would most likely need to support current version and next. How would we deal with the federation.
+* WT: I think it shouldn't be that hard. It sould be to easy to support current and next.
+* AC: To migration path, there are 3 parts to federation when it comes to client identifiers. The client / application interacts with various endponts using client_id query parameter with URL as value. This will not change.
+* ...: The two others are not as easy. On OP level, it will need to use the .well-known instead of just dereferencing the client_id. Check the signature chain etc. It is more complicated. On the plus side, since it is spec in OIDC family of specs. We can expect that OP providers will support it. The Solid specific provider should be able to make use of that. So in the short term there might be more work for OPs but in long term it should be more broadly available.
+* ...: Currently there is not requirement on signatures and trust chains for Client ID document. If we go with OIDC Federation, you can't just publish the document. It has to be signed and there must be trust chain.
+* WT: Do you know about existing implementations of OIDC Federation draft?
+* AC: I currently don't know any. I also don't know about any Solid Client ID outside of Solid.
+* WT: We implement it.
+* AC: I know, I just don't expect adoption outside of Solid-OIDC ecysystem.
+* ES: I understand Wouter talks about use.id
+* eP: For the trust chain, I haven't loooked deeply into that. It would rely heavily on the redirect_uri validation. The basic safety comes from that. What is the minimum needed for the trust/signature chain.
+* AC: In terms of minimal that's needed. On of the big differences between federation and our current client identifiers approach. It mostly relates to federation vs decentralization. As I mentioned currently Client ID Documents don't have to be signed. In a way they are the root of their own federation. How can we replicate that federation of one entity in context of solid. In order to have that minimal structure, there are two concepts: *trust anchor* and *leaf entity*. I understand that they have to be different, IMO it would be easier if they could be the same. At the same time I think both can be provided by the same software package. If we take some app, it could provide both, even created by developer at build time using some static mechanism. So SPA should work as expected. I was unsure if turst anchor has to provide set of APIs, federation API and list API. If we take some photo application, it ends up signing the app key with federation key. The verifier asks for key with some id *1234*. Some problems could come up with relation to implementing it.
```

Comment:
```suggestion
* AC: In terms of minimum that's needed. One of the big differences between federation and our current client identifiers approach mostly relates to federation vs decentralization. As I mentioned, Client ID Documents don't currently have to be signed. In a way, they are the root of their own federation. How can we replicate that federation of one entity in context of Solid? In order to have that minimal structure, there are two concepts: *trust anchor* and *leaf entity*. I understand that they have to be different; IMO, it would be easier if they could be the same. At the same time, I think both can be provided by the same software package. If we take some app, it could provide both, even created by developer at build time using some static mechanism. So SPA should work as expected. I was unsure if trust anchor has to provide set of APIs — federation API and list API. If we take some photo application, it ends up signing the app key with federation key. The verifier asks for key with some id, *1234*. Some problems could come up with relation to implementing it.
```